apps/angular/router-input/src/app/test.component.ts

```diff
@@ -1,22 +1,18 @@
 import { AsyncPipe } from '@angular/common';
-import { Component, inject } from '@angular/core';
-import { ActivatedRoute } from '@angular/router';
-import { map } from 'rxjs';
+import { Component, Input as RouterInput } from '@angular/core';
 
 @Component({
   selector: 'app-subscription',
   standalone: true,
   imports: [AsyncPipe],
   template: `
-    <div>TestId: {{ testId$ | async }}</div>
-    <div>Permission: {{ permission$ | async }}</div>
-    <div>User: {{ user$ | async }}</div>
+    <div>TestId: {{ testId }}</div>
+    <div>Permission: {{ permission }}</div>
+    <div>User: {{ user }}</div>
   `,
 })
 export default class TestComponent {
-  private activatedRoute = inject(ActivatedRoute);
-
-  testId$ = this.activatedRoute.params.pipe(map((p) => p['testId']));
-  permission$ = this.activatedRoute.data.pipe(map((d) => d['permission']));
-  user$ = this.activatedRoute.queryParams.pipe(map((q) => q['user']));
+  @RouterInput() testId!: string;
+  @RouterInput() permission!: string;
+  @RouterInput() user!: string;
```

Comment:
you could try out using the new input syntax, but that's not necessary

Reply:
PR updated to new input syntax ;)